spec/support/shared_contexts/cli/helpers/when_there_are_existing_subgroups_context.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+shared_context 'when there are subgroups which are belong to selected owner' do
+  let(:existing_subgroups_of_selected_owner_group) do
+    [
+      foo_subgroup,
+      bar_subgroup
+    ]
+  end
+
+  let(:foo_subgroup_heading) { '# TopNinjas - WEB Area' }
+  let(:foo_pattern_one) { "foo_one.rb #{default_owner}" }
+  let(:foo_pattern_two) { "foo_two.rb #{default_owner}" }
+  let(:foo_subgroup_list) do
+    [
+      Codeowners::Checker::Group::Comment.new(foo_subgroup_heading),
+      Codeowners::Checker::Group::Pattern.new(foo_pattern_one),
+      Codeowners::Checker::Group::Pattern.new(foo_pattern_two)
+    ]
+  end
```

Comment:
It seems too complicated to keep instantiating one by one. What about we use our own processor to generate the setup?